torch/csrc/cuda/shared/nvtx.cpp

```diff
@@ -1,11 +1,41 @@
 #ifdef _WIN32
 #include <wchar.h> // _wgetenv for nvtx
 #endif
+#include <cuda_runtime.h>
 #include <nvToolsExt.h>
 #include <torch/csrc/utils/pybind.h>
 
 namespace torch::cuda::shared {
 
+struct RangeHandle {
+  nvtxRangeId_t id;
+  const char* msg;
+};
+
+static void flex_callback_range_end(void* userData) {
+  RangeHandle* handle = ((RangeHandle*)userData);
+  nvtxRangeEnd(handle->id);
+  free((void*)handle->msg);
+  free((void*)handle);
```

Comment:
Very much of a detail, which may also depends on conventions that I don't know here:
if `msg` is not marked const in the structure definition above, you should not need the cast.
Unless I miss something, I also think the cast for `handle` itself is not necessary.

Another nitpicking question:
It looks like most of this file is using c++ features over c. Shouldn't we use new/delete and std::string here?

Reply:
@vincent-tr I'll merge as is, let me know if you'd prefer switching to `new`/`delete`/`std::string`: we can do it any time. Alternatively, do not hesitate to submit a PR 😄

Reply:
Was only nitpicking suggestions.
If you feel more comfortable using C-way, then just keep it.